codershq/companies/views.py

```diff
@@ -1,17 +1,50 @@
 from django.urls import reverse_lazy
 from django.views.generic import ListView
+from django.views.generic.base import TemplateView
 from django.views.generic.edit import CreateView, DeleteView, UpdateView, FormView
 from django.contrib.auth.mixins import LoginRequiredMixin
 from codershq.companies.models import Company
-
-
-class CompanyListView(ListView, LoginRequiredMixin):
-    model = Company
-
-
-class CompanyCreateView(CreateView):
-    model = Company
-    fields = '__all__'
+from django.shortcuts import redirect, render
+from .forms import CompanyCreationForm
+from django.core.exceptions import ValidationError
+
+
+class CompanyListView(TemplateView, LoginRequiredMixin):
+    template_name = 'companies/my_companies.html'
+    def get(self, request, *args, **kwargs):
+        
+        exists = Company.objects.filter(user=request.user).exists()
+        if exists:
+            companies = Company.objects.filter(user=request.user)
+            return render(request, self.template_name, {'companies':companies})
+        else:
+            return render(request, self.template_name, {'msg':'No companies to show.'})
+
+class CompanyCreateView(TemplateView, LoginRequiredMixin):
+    template_name = 'companies/company_form.html'
+
+    def get(self, request, *args, **kwargs):
+        if request.user.is_authenticated:
+            if request.user.allowed_to_create_company:
+                return render(request, self.template_name, {'form':CompanyCreationForm})
+            else:
+                return render(request, 'companies/not_allowed.html')
+        else:
+            return render(request, 'pages/welcome.html')
+
+    def post(self, request, *args, **kwargs):
+        form = CompanyCreationForm(request.POST, request.FILES)
+        if form.is_valid():
+            name = request.POST['name']
```

Comment:
isnt there a better way to save the form? I rememebr the form saving itself without you explicitly taking all the values and creating a new object from them.